contracts/crowdsale/validation/TimedCrowdsale.sol

```diff
@@ -0,0 +1,38 @@
+pragma solidity ^0.4.18;
+
+import "../../math/SafeMath.sol";
+import "../Crowdsale.sol";
+
+/**
+ * @dev Crowdsale accepting contributions only within a time frame
+ */
+contract TimedCrowdsale is Crowdsale {
+  using SafeMath for uint256;
+
+  uint256 public startTime;
+  uint256 public endTime;
+
+  /**
+   * @param _startTime Crowdsale opening time
+   * @param _endTime Crowdsale closing time
+   */
+  function TimedCrowdsale(uint256 _startTime, uint256 _endTime) public {
+    require(_startTime >= now);
+    require(_endTime >= _startTime);
+
+    startTime = _startTime;
+    endTime = _endTime;
+  }
+
+  /**
+   * @return Whether crowdsale period has elapsed
+   */
+  function hasExpired() public view returns (bool) {
```

Comment:
Expired instead of Ended (`_endTime`) for any particular reason?

Reply:
`hasEnded` was the prior method that encompassed ending either by reaching `endTime` or by reaching the cap in a `CappedCrowdsale`. Now the purchase validation is done in separate places so we need to split this (`hasExpired` and `capReached`). While we could call this one `hasEnded`, a crowdsale could also end "prematurely" by reaching its cap, thus, be ended but not expired..